### PR TITLE
The ol' spicy keychain

### PR DESCRIPTION
**Pull request by BarneyGumball**
Let's you heat up pretty much any holdable item on the bunsen burner and in both the oven and deep fryer.  The item will then burn whoever proceeds to pick it up without the appropriate protection.  That protection being a pair of highly heat proof gloves.

:cl:
 * rscadd: You can now heat up items on the bunsen burner and in both the oven and deep fryer to play hilarious pranks on your friends!  Be careful though, whatever you're holding will get really hot.  Best you wear something to protect your hands.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
The addition of a new item-level var solely for a meme that can only be performed with bunsen burners seems odd.
I'd rather have a var to properly track the item's actual temperature a la reagent heating, along with a var to determine how quickly it gains/loses heat based on the type of material from which it is made.

**Comment by BarneyGumball** [comment]
I can move the var to an item flag.

I don't like the idea of item temperatures though.  As it is, heating an item functions as a quick little "haha" moment, at which point you can go back to using the item.  Having proper item heating/cooling has much longer lasting implications for any items that get heated up.  I can only imagine the nightmare after a plasma fire.  Also, having every `/obj` in the world process temperature changes probably wouldn't be worth the performance cost.

**Comment by SGuivenne** [comment]
only if we can heat up the chain of command and then burn people with it.   I also expect this to eventually be used to brand people as punishment for crimes.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I imagine the best way performance-wise to implement this would be an on-pickup event but I'm still not convinced adding a single var is that bad

Also, what if spicy items had a red outline like staff of reanimation mobs?

**Comment by kilozombie** [comment]
> I imagine the best way performance-wise to implement this would be an on-pickup event but I'm still not convinced adding a single var is that bad
> 
> Also, what if spicy items had a red outline like staff of reanimation mobs?

There would have to be a tiny window where it's spicy hot but not visibly hot so you can really give somebody the ol' spicy cryptographic sequencer

**Comment by kilozombie** [comment]
Also, steal the Duck Game "the item you're holding is red hot" hissing sound effect

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Excuse me in the original spicy keychain picture the keys are visibly red

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
What the vox person said. And, for goodness sake don't make it just a bunsen burner.

**Comment by BarneyGumball** [comment]
>don't make it just a bunsen burner

I'm open to suggestions.

**Comment by kilozombie** [comment]
Have it be a function in the oven, like in the original

**Comment by BarneyGumball** [comment]
Alright, so I:
* Adjusted the contents of some messages.
* Made it so you can heat up items in the oven and the deep fryer.
* Fixed the gloves protection so you can't just remove your gloves after you've picked up the heated item.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -807,6 +810,53 @@
...
+		damage_applied = user.apply_damage(5, BURN, hand_used)
+	// Spit out message based on if they were burned and/or they feel pain.
+	// Also makes them drop the item if they're holding it and feel pain
+	if(damage_applied)
+		if(user.feels_pain())
+			if (hand_index) // they're holding the item, make them drop it
+				user.u_equip(src, 1)
+			user.audible_scream()
+			user.visible_message(
+					"<span class='warning'>[user] cries out in pain and jerks \his [hand_used] \
```
Comment (BarneyGumball) [inline code/game/objects/items.dm:843]: I don't actually know if `[hand_used]` works here, pls advise.

Edit: Ok it should work.  Not sure how to test it on my own though.

**Comment by MadmanMartian** [comment]
So, added a proc for you to use. Attempt_heating. It's object level, takes an argument of the object being used to heat it, and the user if available.

Objects have is_hot() which returns the temperature, and thermal_energy_transfer() which returns the energy transfer

Use that.

**Comment by BarneyGumball** [comment]
This pr SUCKS

**Pull request closed without merging**